packages/grid/x-data-grid/src/components/DataGridVirtualScroller.tsx

```diff
@@ -4,24 +4,30 @@ import { GridVirtualScrollerContent } from './virtualization/GridVirtualScroller
 import { GridVirtualScrollerRenderZone } from './virtualization/GridVirtualScrollerRenderZone';
 import { useGridVirtualScroller } from '../hooks/features/virtualization/useGridVirtualScroller';
 import { GridOverlays } from './base/GridOverlays';
+import { GridHeaders } from './GridHeaders';
 
-interface DataGridVirtualScrollerProps extends React.HTMLAttributes<HTMLDivElement> {
-  disableVirtualization?: boolean;
+export interface DataGridVirtualScrollerProps extends React.HTMLAttributes<HTMLDivElement> {
+  ref: React.Ref<HTMLDivElement>;
+  disableVirtualization: boolean;
+  ColumnHeadersProps?: Record<string, any>;
 }
 
 const DataGridVirtualScroller = React.forwardRef<HTMLDivElement, DataGridVirtualScrollerProps>(
   function DataGridVirtualScroller(props, ref) {
-    const { className, disableVirtualization, ...other } = props;
+    const { className, disableVirtualization, ColumnHeadersProps, ...other } = props;
 
     const { getRootProps, getContentProps, getRenderZoneProps, getRows } = useGridVirtualScroller({
       ref,
       disableVirtualization,
     });
 
+    const contentProps = getContentProps();
+
     return (
       <GridVirtualScroller className={className} {...getRootProps(other)}>
+        <GridHeaders contentProps={contentProps} ColumnHeadersProps={ColumnHeadersProps} />
```

Comment:
The cell keyboard navigation will need to take the header height into account. Currently, the focused cell hides below the fold:

https://github.com/mui/mui-x/assets/13808724/dfe6bbba-4dda-474b-80f0-3a2103bc758e